airbyte-integrations/connectors/source-airtable/source_airtable/source.py

```diff
@@ -3,99 +3,77 @@
-        super().__init__(**kwargs)
-        self.base_id = base_id
-        self.table_name = table_name
-        self.schema = schema
-
-    @property
-    def name(self):
-        return self.table_name
-
-    def get_json_schema(self) -> Mapping[str, Any]:
-        return self.schema
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        json_response = response.json()
-        offset = json_response.get("offset", None)
-        if offset:
-            return {"offset": offset}
-        return None
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        if next_page_token:
-            return next_page_token
-        return {}
-
-    def process_records(self, records):
-        for record in records:
-            data = record.get("fields", {})
-            processed_record = {"_airtable_id": record.get("id"), "_airtable_created_time": record.get("createdTime"), **data}
-            yield processed_record
-
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        json_response = response.json()
-        records = json_response.get("records", [])
-        records = self.process_records(records)
-        yield from records
-
-    def path(
-        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> str:
-        return f"{self.base_id}/{self.table_name}"
+class SourceAirtable(AbstractSource):
 
+    logger: logging.Logger = logging.getLogger("airbyte")
+    streams_catalog: Iterable[Mapping[str, Any]] = []
 
-# Source
-class SourceAirtable(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         auth = TokenAuthenticator(token=config["api_key"])
-        for table in config["tables"]:
-            try:
-                Helpers.get_most_complete_row(auth, config["base_id"], table)
-            except Exception as e:
-                return False, str(e)
-        return True, None
+        try:
+            # try reading first table from each base, to check the connectivity,
+            for base in AirtableBases(authenticator=auth).read_records(sync_mode=None):
+                base_id = base.get("id")
+                base_name = base.get("name")
+                self.logger.info(f"Reading first table info for base: {base_name}")
+                next(AirtableTables(base_id=base_id, authenticator=auth).read_records(sync_mode=None))
+            return True, None
+        except Exception as e:
+            return False, str(e)
+
+    def prepare_catalog_for_base(self, base_id: str, base_name: str, base_tables: list) -> Iterable[Mapping[str, Any]]:
```

Comment:
The type hint of this function mentions we can expect it to return an iterable, but it actually does not return anything and mutate `self.streams`. I'd prefer it to produce a list of Airbyte Streams, and you make discover call this function to get the list.
```suggestion
    def get_streams_from_base(self, base_id: str, base_name: str, base_tables: list) -> Iterable[Mapping[str, Any]]:
```

Reply:
Now the logic populates ready-streams into the list, and processes that list, I've moved this logic into SchemaHelpers to clear the code, but would rather avoid returning the list from `get_streams_from_base` method and leave it as is.
the TypeHint was removed as well.